tiledb/sm/misc/types.h

```diff
@@ -260,14 +260,108 @@ class Range {
 /** An N-dimensional range, consisting of a vector of 1D ranges. */
 typedef std::vector<Range> NDRange;
 
-/** A value as a vector of bytes. */
-typedef std::vector<uint8_t> ByteVecValue;
+/** An untyped value, barely more than raw storage. This class is only
+ * transitional. All uses should be rewritten to use ordinary types. Consider
+ * it deprecated at creation.
+ *
+ * This class started off as a typedef for a byte vector. In its current state,
+ * it provides methods that capture common patterns of usage, avoiding bleeding
+ * all its abstraction into calling code. It's not perfect, and never will be.
+ *
+ * A minimal number of vector methods are forwarded outside the class to allow
+ * not-yet-converted code its legacy behavior. The incremental goal is to remove
+ * such functions as the code base evolves away from untyped variables entirely.
+ */
+
+class ByteVecValue {
+  typedef std::vector<uint8_t> Base;
+  std::vector<uint8_t> x;
```

Comment:
I think the tiledb convention is to always use underscore suffixed member variable names, i.e. 'x_'.

Reply:
Agreed this is the convention elsewhere in the code.